instrumentation/ktor/ktor-2-common/library/src/main/kotlin/io/opentelemetry/instrumentation/ktor/v2_0/common/AbstractKtorClientTelemetryBuilder.kt

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.ktor.v2_0.common
+
+import io.ktor.client.request.*
+import io.ktor.client.statement.*
+import io.ktor.http.*
+import io.opentelemetry.api.OpenTelemetry
+import io.opentelemetry.api.common.AttributesBuilder
+import io.opentelemetry.context.Context
+import io.opentelemetry.instrumentation.api.incubator.builder.internal.DefaultHttpClientInstrumenterBuilder
+import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor
+import io.opentelemetry.instrumentation.api.instrumenter.SpanNameExtractor
+import io.opentelemetry.instrumentation.ktor.v2_0.common.internal.KtorBuilderUtil
+import java.util.function.Function
+
+abstract class AbstractKtorClientTelemetryBuilder(
+  private val instrumentationName: String
+) {
+  companion object {
+    init {
+      KtorBuilderUtil.clientBuilderExtractor = { it.builder }
+    }
+  }
+
+  internal lateinit var openTelemetry: OpenTelemetry
+  internal lateinit var internalBuilder: DefaultHttpClientInstrumenterBuilder<HttpRequestData, HttpResponse>
+  protected lateinit var builder: DefaultHttpClientInstrumenterBuilder<HttpRequestData, HttpResponse>
```

Comment:
I couldn't figure out how to have both internal and protected access

Reply:
`internal` should be enough -- is there a reason why it shouldn't be public? It can only be used by other classes in this module when it's `internal`. 👍🏻

Reply:
I found a better way than duplicating the field, I created an `internal` accessor to the `protected` field